draft-dekater-scion-controlplane.md

```diff
@@ -1297,6 +1309,64 @@ In comparison to these time scales, clock offsets in the order of minutes are im
 Each administrator of a SCION control service is responsible for maintaining sufficient clock accuracy. No particular method is assumed by this specification.
 
 
+## Path Discovery Time and Scalability
+
+<!-- TODO: sentence -->
+Unreachable goal: discover all policy compliant paths, ~immediately
+Resource trade off: reduce propagation interval, reduce candidate set -> not all paths are found, takes some time.
+
+This resource "costs" for path discovery are communication overhead, processing and storage. Communication is transmitting the PCBs and occasionally obtaining the required PKI material. Processing cost is validating the signatures of the AS entries, signing new AS entries, and, to a lesser extent, evaluating the beaconing policies.
+All of these depend on the the number and length of the discovered path segments, that is, on the total number of AS entries of the discovered path segments.
+
+Interesting metrics for scalability and speed of path discovery are the time until all discoverable path segments have been discovered after a "cold boot" and the time until a new link is usable.
+<!-- TODO: Note: removing links is not the job of the path discovery. For scheduled removal, path with old links expire. For link failures, paths are not renewed, routing around failure by path choice (data plane) -->
+Generally, the time until a specific PCB is built depends on its length and the propagation interval.
+At each AS, the PCB will be processed and propagated at the subsequent propagation interval.
+With a propagation interval T, the mean time until the PCB is propgated in one AS is T / 2 and the mean total time for the propagation steps of a PCB of length L is L * T / 2 (with a variance of L * T^2 / 12).
```

Comment:
Here I'd cite the Irwin-Hall distribution, may be?

Reply:
It seems Irwin-Hall was the wrong pointer after all.
Generally, the variance of the sum of independent random variables is just the sum of the variances;  the variance of the sum of n independent, _identically_ distributed variables is just n times the variance of a single variable.
The variance of continuously uniformly distributed random variables over interval [a,b] is (b-a)^/12. (!) -- in our case, this interval is [0,T] and so the variance for the individual random variables is T^2/12.

Even if this was not something I knew directly, this seems too basic / too much textbook knowledge to add a citation or long explanation. What do you think.

Reply:
.. which means that we don't actual care about T^2 / 12, do we? Anyway, you're right. The stats heads will know and the others won't care or understand (you just demonstrated this on me: I didn't understand).